### PR TITLE
[16.0] [MIG] resource_work_time_from_contracts

### PR DESCRIPTION
**Pull request by victor-champonnois**
@huguesdk I have one failing test and I can't make it work. Also I am not sure about commit eca18007f3626068cace513d2160a0f06324e7f3.
Could you help ?

The error happens for `test_with_leaves`

When I stop at line https://github.com/coopiteasy/addons/blob/eca18007f3626068cace513d2160a0f06324e7f3/resource_work_time_from_contracts/models/resource_mixin.py#L143 and do `self.list_leaves(from_datetime,to_datetime)`
```
0:
(datetime.date(2021, 10, 26), 1.8, resource.calendar.leaves(81,))
1:
(datetime.date(2021, 10, 26), 1.0, resource.calendar.leaves(81,))
2:
(datetime.date(2021, 10, 27), 1.8, resource.calendar.leaves(82,))
3:
(datetime.date(2021, 10, 29), 1.8, resource.calendar.leaves(83,))
4:
(datetime.date(2021, 10, 29), 3.8, resource.calendar.leaves(83,))
````
Alors qu'en 12 j'obtiens :
`````
0:
(datetime.date(2021, 10, 26), 3.8, resource.calendar.leaves(10,))
1:
(datetime.date(2021, 10, 27), 3.8, resource.calendar.leaves(11,))
2:
(datetime.date(2021, 10, 29), 3.8, resource.calendar.leaves(12,))
3:
(datetime.date(2021, 10, 29), 3.8, resource.calendar.leaves(12,))
```

normal_work_time_per_day est OK. 

task : https://gestion.coopiteasy.be/web#id=10959&action=475&active_id=492&model=project.task&view_type=form&menu_id=

### CODE REVIEW

File: resource_work_time_from_contracts/models/resource_mixin.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def list_normal_work_time_per_day(self, from_datetime, to_datetime, domain=None)
             day += delta
         return result
 
-    def get_work_days_data(
+    def _get_work_days_data_batch(
```
Comment (carmenbianca) [inline resource_work_time_from_contracts/models/resource_mixin.py]: Per the commit message, which indirectly links to https://github.com/odoo/odoo/pull/56477, I think this change (and all other changes in the commit 7bbb859bc545e39a50980867077177e30714badf) should be squashed into the 13.0 migration commit.

File: resource_work_time_from_contracts/tests/test_work_days_data.py [hunk 2]
```diff
@@ -164,8 +165,8 @@ def test_with_leaves(self):
             {
                 "name": "Tuesday morning",
                 "calendar_id": self.employee1.resource_calendar_id.id,
-                "date_from": self.to_utc_datetime(2021, 10, 26, 8, 42),
-                "date_to": self.to_utc_datetime(2021, 10, 26, 12, 30),
+                "date_from": datetime_str(2021, 10, 26, 8, 42),
+                "date_to": datetime_str(2021, 10, 26, 12, 30),
```
Comment (carmenbianca) [inline resource_work_time_from_contracts/tests/test_work_days_data.py]: I'm not sure why this choice was made. Wouldn't it be more intuitive to type `"2021-10-26T08:42"`? Same for all others.

**Review by carmenbianca — CHANGES_REQUESTED** [review]
Some small comments.

Will also look at the failing test.

File: resource_work_time_from_contracts/tests/test_work_days_data.py (hunk 2 above)
Comment (carmenbianca) [reply resource_work_time_from_contracts/tests/test_work_days_data.py]: no longer relevant

**Comment by carmenbianca** [comment]
Patched up the failing test. Please remove the reverted commit (and the revert commit) before merging. I kept them for history's sake.

File: resource_work_time_from_contracts/models/resource_mixin.py (hunk 1 above)
Comment (victor-champonnois) [reply resource_work_time_from_contracts/models/resource_mixin.py]: done

**Comment by victor-champonnois** [comment]
thanks a lot carmen !

**Review by carmenbianca — APPROVED** [review]

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**